Serde macro namespace clash for Windows builds

I am using both `serde` and `serde_derive` in my own crate which depends on `hdf5`. After adding the two dependencies to my crate, the following happens:

```
error[E0252]: the name `Deserialize` is defined multiple times
   --> lib\hdf5-rust\hdf5-sys\build.rs:345:9
    |
344 |     use serde::{Deserialize, Deserializer};
    |                 ----------- previous import of the macro `Deserialize` here
345 |     use serde_derive::Deserialize;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^ `Deserialize` reimported here
    |
    = note: `Deserialize` must be defined only once in the macro namespace of this module
```

This prevents me from successfully building the library, but the fix is rather easy:

```rust
use serde::{Deserialize, Deserializer};
use serde_derive::{Deserialize as SDDeserialize}; 
// [...]
#[derive(Clone, SDDeserialize)]
struct App {
```

I found [this issue](https://github.com/serde-rs/serde/issues/1441) which seems related.

--- a/hdf5-sys/build.rs
+++ b/hdf5-sys/build.rs
@@ -342,7 +342,7 @@ mod windows {
 
     use serde::de::Error;
     use serde::{Deserialize, Deserializer};
-    use serde_derive::Deserialize;
+    use serde_derive::Deserialize as DeriveDeserialize;
     use winreg::enums::HKEY_LOCAL_MACHINE;
     use winreg::RegKey;
 
@@ -356,7 +356,7 @@ mod windows {
         }
     }
 
-    #[derive(Clone, Deserialize)]
+    #[derive(Clone, DeriveDeserialize)]
     struct App {
         #[serde(rename = "DisplayName")]
         name: String,

--- a/src/hl/dataset.rs
+++ b/src/hl/dataset.rs
@@ -5,12 +5,12 @@ use std::ops::Deref;
 use num_integer::div_floor;
 
 #[cfg(hdf5_1_10_5)]
-use hdf5_sys::h5d::H5Dget_chunk_info;
+use hdf5_sys::h5d::{H5Dget_chunk_info, H5Dget_num_chunks};
 use hdf5_sys::{
     h5::HADDR_UNDEF,
     h5d::{
         H5D_fill_value_t, H5D_layout_t, H5Dcreate2, H5Dcreate_anon, H5Dget_create_plist,
-        H5Dget_num_chunks, H5Dget_offset, H5Dset_extent, H5D_FILL_TIME_ALLOC,
+        H5Dget_offset, H5Dset_extent, H5D_FILL_TIME_ALLOC,
     },
     h5p::{
         H5Pcreate, H5Pfill_value_defined, H5Pget_chunk, H5Pget_fill_value, H5Pget_layout,
@@ -105,6 +105,7 @@ impl Dataset {
         })
     }
 
+    #[cfg(hdf5_1_10_5)]
     /// Returns number of chunks if the dataset is chunked.
     pub fn num_chunks(&self) -> Option<usize> {
         if !self.is_chunked() {

--- a/src/hl/dataset.rs
+++ b/src/hl/dataset.rs
@@ -4,6 +4,8 @@ use std::ops::Deref;
 
 use num_integer::div_floor;
 
+#[cfg(hdf5_1_10_5)]
+use hdf5_sys::h5d::H5Dget_chunk_info;
 use hdf5_sys::{
     h5::HADDR_UNDEF,
     h5d::{
@@ -62,6 +64,32 @@ pub enum Chunk {
     Manual(Vec<Ix>),
 }
 
+#[cfg(hdf5_1_10_5)]
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ChunkInfo {
+    /// Array with a size equal to the dataset’s rank whose elements contain 0-based
+    /// logical positions of the chunk’s first element in each dimension.
+    pub offset: Vec<u64>,
+    /// Filter mask that indicates which filters were used with the chunk when written.
+    /// A zero value indicates that all enabled filters are applied on the chunk.
+    /// A filter is skipped if the bit corresponding to the filter’s position in
+    /// the pipeline (0 ≤ position < 32) is turned on.
+    pub filter_mask: u32,
+    /// Chunk address in the file.
+    pub addr: u64,
+    /// Chunk size in bytes.
+    pub size: u64,
+}
+
+#[cfg(hdf5_1_10_5)]
+impl ChunkInfo {
+    pub(crate) fn new(ndim: usize) -> Self {
+        let mut offset = Vec::with_capacity(ndim);
+        unsafe { offset.set_len(ndim) };
+        Self { offset, filter_mask: 0, addr: 0, size: 0 }
+    }
+}
+
 impl Dataset {
     /// Returns whether this dataset is resizable along some axis.
     pub fn is_resizable(&self) -> bool {
@@ -85,6 +113,28 @@ impl Dataset {
         h5lock!(self.space().map_or(None, |s| {
             let mut n: hsize_t = 0;
             h5check(H5Dget_num_chunks(self.id(), s.id(), &mut n)).map(|_| n as _).ok()
+        }))
+    }
+
+    #[cfg(hdf5_1_10_5)]
+    /// Retrieves the chunk information for the chunk specified by its index.
+    pub fn chunk_info(&self, index: usize) -> Option<ChunkInfo> {
+        if !self.is_chunked() {
+            return None;
+        }
+        h5lock!(self.space().map_or(None, |s| {
+            let mut chunk_info = ChunkInfo::new(self.ndim());
+            h5check(H5Dget_chunk_info(
+                self.id(),
+                s.id(),
+                index as _,
+                chunk_info.offset.as_mut_ptr(),
+                &mut chunk_info.filter_mask,
+                &mut chunk_info.addr,
+                &mut chunk_info.size,
+            ))
+            .map(|_| chunk_info)
+            .ok()
         }))
     }
 

--- a/src/hl/dataset.rs
+++ b/src/hl/dataset.rs
@@ -8,7 +8,7 @@ use hdf5_sys::{
     h5::HADDR_UNDEF,
     h5d::{
         H5D_fill_value_t, H5D_layout_t, H5Dcreate2, H5Dcreate_anon, H5Dget_create_plist,
-        H5Dget_offset, H5Dset_extent, H5D_FILL_TIME_ALLOC,
+        H5Dget_num_chunks, H5Dget_offset, H5Dset_extent, H5D_FILL_TIME_ALLOC,
     },
     h5p::{
         H5Pcreate, H5Pfill_value_defined, H5Pget_chunk, H5Pget_fill_value, H5Pget_layout,
@@ -75,6 +75,17 @@ impl Dataset {
                 .ok()
                 .map_or(false, |dcpl_id| H5Pget_layout(dcpl_id) == H5D_layout_t::H5D_CHUNKED)
         })
+    }
+
+    /// Returns number of chunks if the dataset is chunked.
+    pub fn num_chunks(&self) -> Option<usize> {
+        if !self.is_chunked() {
+            return None;
+        }
+        h5lock!(self.space().map_or(None, |s| {
+            let mut n: hsize_t = 0;
+            h5check(H5Dget_num_chunks(self.id(), s.id(), &mut n)).map(|_| n as _).ok()
+        }))
     }
 
     /// Returns the chunk shape if the dataset is chunked.

--- a/src/hl/datatype.rs
+++ b/src/hl/datatype.rs
@@ -4,12 +4,12 @@ use std::fmt::{self, Debug, Display};
 use std::ops::Deref;
 
 use hdf5_sys::h5t::{
-    H5T_cdata_t, H5T_class_t, H5T_cset_t, H5T_str_t, H5Tarray_create2, H5Tcompiler_conv, H5Tcopy,
-    H5Tcreate, H5Tenum_create, H5Tenum_insert, H5Tequal, H5Tfind, H5Tget_array_dims2,
-    H5Tget_array_ndims, H5Tget_class, H5Tget_cset, H5Tget_member_name, H5Tget_member_offset,
-    H5Tget_member_type, H5Tget_member_value, H5Tget_nmembers, H5Tget_sign, H5Tget_size,
-    H5Tget_super, H5Tinsert, H5Tis_variable_str, H5Tset_cset, H5Tset_size, H5Tset_strpad,
-    H5Tvlen_create, H5T_VARIABLE,
+    H5T_cdata_t, H5T_class_t, H5T_cset_t, H5T_order_t, H5T_str_t, H5Tarray_create2,
+    H5Tcompiler_conv, H5Tcopy, H5Tcreate, H5Tenum_create, H5Tenum_insert, H5Tequal, H5Tfind,
+    H5Tget_array_dims2, H5Tget_array_ndims, H5Tget_class, H5Tget_cset, H5Tget_member_name,
+    H5Tget_member_offset, H5Tget_member_type, H5Tget_member_value, H5Tget_nmembers, H5Tget_order,
+    H5Tget_sign, H5Tget_size, H5Tget_super, H5Tinsert, H5Tis_variable_str, H5Tset_cset,
+    H5Tset_size, H5Tset_strpad, H5Tvlen_create, H5T_VARIABLE,
 };
 use hdf5_types::{
     CompoundField, CompoundType, EnumMember, EnumType, FloatSize, H5Type, IntSize, TypeDescriptor,
@@ -119,10 +119,49 @@ impl Default for Conversion {
     }
 }
 
+#[derive(Copy, Debug, Clone, PartialEq, Eq)]
+pub enum ByteOrder {
+    LittleEndian,
+    BigEndian,
+    Vax,
+    Mixed,
+    None,
+}
+
+#[cfg(hdf5_1_8_6)]
+impl From<H5T_order_t> for ByteOrder {
+    fn from(order: H5T_order_t) -> Self {
+        match order {
+            H5T_order_t::H5T_ORDER_LE => ByteOrder::LittleEndian,
+            H5T_order_t::H5T_ORDER_BE => ByteOrder::BigEndian,
+            H5T_order_t::H5T_ORDER_VAX => ByteOrder::Vax,
+            H5T_order_t::H5T_ORDER_MIXED => ByteOrder::Mixed,
+            _ => ByteOrder::None,
+        }
+    }
+}
+
+#[cfg(not(hdf5_1_8_6))]
+impl From<H5T_order_t> for ByteOrder {
+    fn from(order: H5T_order_t) -> Self {
+        match order {
+            H5T_order_t::H5T_ORDER_LE => ByteOrder::LittleEndian,
+            H5T_order_t::H5T_ORDER_BE => ByteOrder::BigEndian,
+            H5T_order_t::H5T_ORDER_VAX => ByteOrder::Vax,
+            _ => ByteOrder::None,
+        }
+    }
+}
+
 impl Datatype {
     /// Get the total size of the datatype in bytes.
     pub fn size(&self) -> usize {
         h5call!(H5Tget_size(self.id())).unwrap_or(0) as usize
+    }
+
+    /// Get the byte order of the datatype.
+    pub fn byte_order(&self) -> ByteOrder {
+        h5lock!(H5Tget_order(self.id())).into()
     }
 
     pub fn conv_path<D>(&self, dst: D) -> Option<Conversion>

